src/main/java/edu/agh/wfiis/designPatterns/comparator/DesignPatternComparator.java

```diff
@@ -0,0 +1,11 @@
+package edu.agh.wfiis.designpatterns.designPatterns.comparator;
+
+import edu.agh.wfiis.designpatterns.designPatterns.pattern.DesignPattern;
+
+public class DesignPatternComparator {
+
+    public boolean areSameType(DesignPattern first, DesignPattern second){
+        return first.getType() == second.getType();
```

Comment:
Przy porównywaniu zmiennych `String` używamy raczej metody `equals` niż operatora `==`, bo `==` porównuje referencje, dwa różne obiekty tego samego patternu zwrócą tutaj false.

Reply:
Pattern is of type enum which is guaranteed to have no more than one instance